### ISSUE TITLE
Replacing custom_vjp on latitudinal step with custom JAX primitives (custom_vjp lacks a transpose)

### ISSUE THREAD

**Issue by ArtemBasyrov** [astro-informatics/s2fft#379]
The latitudinal Wigner-d recursion in s2fft/transforms/spherical.py is wrapped in jax.custom_vjp. This works for jax.grad, but custom_vjp does not declare a transpose rule, so jax.linear_transpose cannot peel through the latitudinal step — even though the step is linear and has an exact analytical transpose (the forward step is the transpose of the inverse step and vice versa).

I tried to fix it with jax.experimental.custom_transpose,but unfortunately I couldn't find this API (using JAX 0.10.0). 

Instead I propose solving it by registering the forward and inverse latitudinal steps as custom JAX primitives similar to ffts in s2fft/utils/healpix_ffts.py with an explicit transpose rule (plus a JVP rule and a batcher), instead of custom_vjp. I have my solution to this issue on my fork. 

**The idea**

jax.custom_vjp declares a VJP — fwd / bwd — but does not declare how to transpose the computation as a linear map. Any caller that goes through jax.linear_transpose (directly, or indirectly via JAX transforms that lower to it) cannot transpose the latitudinal step analytically and will fail. For example, when calling the transpose of s2fft.transforms.spherical.inverse_jax.

The operation is linear, so jax.linear_transform should be the API to call, however it fails without custom transpose rule declared. 

**Proposed solution**

Register two custom JAX primitives — flm_to_ftm (inverse latitudinal step) and ftm_to_flm (forward latitudinal step) — using the existing s2fft.utils.jax_primitive.register_primitive helper that's already used for the HEALPix CUDA targets.

Each primitive declares:

- abstract eval — output shape/dtype from samples.ftm_shape samples.flm_shape,
- MLIR lowering — mlir.lower_fun of the existing otf.*_latitudinal_step_jax (numerics unchanged),
- JVP rule — linear ⇒ tangent is the primitive re-bound on the tangent input,
- explicit transpose rule — the actual point of this change: the transpose of flm_to_ftm is
ftm_to_flm, and vice versa,
- batching rule — lifts every operand to a shared leading batch dim and re-binds the primitive, so the analytical-transpose path survives vmap.

Sketch of the transpose rule, which is the part that solves jax.linear_transpose:

```
def _flm_to_ftm_transpose(cotangent, flm, thetas, spin, *precomps,
                          L, nside, sampling, reality, spmd, L_lower):
    # Analytical adjoint: transpose of the inverse step IS the forward step.
    # `precomps` are direction-specific, so pass None to regenerate the
    # forward-direction coefficients.
    cot_flm = _underlying_forward(
        cotangent, thetas, spin, None,
        L=L, nside=nside, sampling=sampling,
        reality=reality, spmd=spmd, L_lower=L_lower,
    )
    return (cot_flm, None, None) + (None,) * len(precomps)

_flm_to_ftm_primitive = register_primitive(
    "flm_to_ftm",
    multiple_results=False,
    abstract_evaluation=_flm_to_ftm_abstract,
    lowering_per_platform={None: mlir.lower_fun(_flm_to_ftm_impl,
                                                multiple_results=False)},
    batcher=_flm_to_ftm_batcher,
    jacobian_vector_product=_flm_to_ftm_jvp,
    transpose=_flm_to_ftm_transpose,
    is_linear=False,  # marked False because spin/precomps are operands
)
```

Call sites in spherical.py collapse to a single function call replacing the custom_vjp

```
ftm = ftm_flm_prim.flm_to_ftm(
    flm, thetas, L=L, spin=spin, nside=nside, sampling=sampling,
    reality=reality, spmd=spmd, L_lower=L_lower, precomps=precomps,
)
```

With this in place, jax.linear_transpose succeeds and routes through the analytical adjoint instead of failing.
